spec/unit/util_spec.rb

```diff
@@ -229,11 +229,29 @@ def get_mode(file)
       (0..256).each {|n| IO.stubs(:new) }
     end
 
-    it "should close all open file descriptors except stdin/stdout/stderr" do
+    it "should close all open file descriptors except stdin/stdout/stderr when /proc/self/fd exists" do
+      # This is ugly, but I can't really think of a better way to do it without
+      # letting it actually close fds, which seems risky
+      [".","..",0,1,2].each {|n| IO.expects(:new).with(n).never}
+      fds = [".", "..","0","1","2","3","5","100","1000"]
+      Dir.stubs(:foreach).with('/proc/self/fd').multiple_yields(*fds)
+      fds.delete(".")
```

Comment:
I think we can remove the `delete` calls in favor of the following in the each lambda:

Note: not tested code follows:

```ruby
fds.each do |fd|
  if fd in ['.', '..', '0', '1', '2']
    IO.expects(:new).with(n).never if 
  else
    IO.expects(:new).with(n).returns mock('io', :close)
  end
end
```